chain-impl-mockchain/src/evm/mod.rs

```diff
@@ -48,66 +47,132 @@ pub enum EvmTransaction {
                 access_list,
             } => {
-                // Set Transaction type
-                let bb = _bb.u8(0);
-                let bb = serialize_address(bb, caller);
-                let bb = serialize_u256(bb, value);
-                let bb = serialize_bytecode(bb, init_code);
-                let bb = serialize_gas_limit(bb, gas_limit);
-                serialize_access_list(bb, access_list)
+                s.begin_list(6);
+                s.append(&u8::from(self));
+                s.append(caller);
+                s.append(value);
+                s.append(init_code);
+                s.append(gas_limit);
+                s.append_list(access_list);
             }
-            #[cfg(feature = "evm")]
-            EvmTransaction::Create2 {
+            Create2 {
                 caller,
                 value,
                 init_code,
                 salt,
                 gas_limit,
                 access_list,
             } => {
-                let bb = _bb.u8(1);
-                let bb = serialize_address(bb, caller);
-                let bb = serialize_u256(bb, value);
-                let bb = serialize_bytecode(bb, init_code);
-                let bb = serialize_h256(bb, salt);
-                let bb = serialize_gas_limit(bb, gas_limit);
-                serialize_access_list(bb, access_list)
+                s.begin_list(7);
+                s.append(&u8::from(self));
+                s.append(caller);
+                s.append(value);
+                s.append(init_code);
+                s.append(salt);
+                s.append(gas_limit);
+                s.append_list(access_list);
             }
-            #[cfg(feature = "evm")]
-            EvmTransaction::Call {
+            Call {
                 caller,
                 address,
                 value,
                 data,
                 gas_limit,
                 access_list,
             } => {
-                let bb = _bb.u8(2);
-                let bb = serialize_address(bb, caller);
-                let bb = serialize_address(bb, address);
-                let bb = serialize_u256(bb, value);
-                let bb = serialize_bytecode(bb, data);
-                let bb = serialize_gas_limit(bb, gas_limit);
-                serialize_access_list(bb, access_list)
+                s.begin_list(7);
+                s.append(&u8::from(self));
+                s.append(caller);
+                s.append(address);
+                s.append(value);
+                s.append(data);
+                s.append(gas_limit);
+                s.append_list(access_list);
             }
-            #[cfg(not(feature = "evm"))]
-            _ => unreachable!(),
+        }
+    }
+}
+
+impl EvmTransaction {
+    /// Serialize the contract into a `ByteBuilder`.
+    pub fn serialize_in(&self, _bb: ByteBuilder<Self>) -> ByteBuilder<Self> {
+        #[cfg(feature = "evm")]
+        {
+            _bb.bytes(&self.rlp_bytes())
```

Comment:
```suggestion
            let bytes = &self.rlp_bytes();
            _bb.u64(bytes.len() as u64).bytes(&bytes)
```

Reply:
Thanks. RLP is the encoding that Ethereum uses, in hashes that are 256-bit long. I think that `u64` len should be enough.